src/compiler/checker.ts

```diff
@@ -21275,6 +21275,8 @@ namespace ts {
                         (isBinaryExpression(target) && target.operatorToken.kind === SyntaxKind.CommaToken && isMatchingReference(source, target.right));
             }
             switch (source.kind) {
+                case SyntaxKind.MetaProperty: /* WIP */
+                    return target.kind === SyntaxKind.MetaProperty;

```

Comment:
This probably needs to do some symbol resolving, but I couldn't find how to do that.